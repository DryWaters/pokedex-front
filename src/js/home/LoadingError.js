import React from 'react';
import PropTypes from 'prop-types';

import '../../styles/home/LoadingError.css';

const propTypes = {
  error: PropTypes.string.isRequired,
}

class LoadingError extends React.Component {
  constructor (props) {
    super(props);

    this.showError = this.showError.bind(this);
  }

  showError (evt) {
    evt.preventDefault();
    alert(this.props.error);
  }

  render () {
    return (
      <h2>
        {'Oh no! There was an issue loading Pokemon! '}
        <button className="show-error-button" onClick={this.showError} >
          {'See error'}
        </button>
      </h2>
    );
  }
}

LoadingError.propTypes = propTypes;

export default LoadingError;
