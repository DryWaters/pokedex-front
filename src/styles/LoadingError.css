/*From https://stackoverflow.com/questions/1367409/how-to-make-button-look-like-a-link*/
.show-error-button {
     background: none!important;
     color: inherit;
     border: none;
     padding: 0!important;
     font: inherit;
     text-decoration: underline;
     cursor: pointer;
}
