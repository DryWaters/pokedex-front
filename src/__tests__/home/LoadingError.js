import React from 'react';
import { shallow } from 'enzyme';

import LoadingError from '../../js/home/LoadingError';

describe('LoadingError Component', () => {
  it('should render LoadingError component without crashing', () => {
    shallow(<LoadingError error="test error" />);
  });

  it('should match LoadingError component snapshot', () => {
    const wrapper = shallow(<LoadingError error="test error" />);
    expect(wrapper).toMatchSnapshot();
  });
});
