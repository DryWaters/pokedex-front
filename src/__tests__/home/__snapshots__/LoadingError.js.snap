// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingError Component should match LoadingError component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingError
    error="test error"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Oh no! There was an issue loading Pokemon! ",
        <button
          className="show-error-button"
          onClick={[Function]}
>
          See error
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Oh no! There was an issue loading Pokemon! ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "See error",
          "className": "show-error-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "See error",
        "type": "button",
      },
    ],
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Oh no! There was an issue loading Pokemon! ",
          <button
            className="show-error-button"
            onClick={[Function]}
>
            See error
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Oh no! There was an issue loading Pokemon! ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "See error",
            "className": "show-error-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "See error",
          "type": "button",
        },
      ],
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
